### PR TITLE
#515: Fix issues with mixed apiGroup and apiVersion.

### PR DESCRIPTION
**Pull request by iocanel**

### CODE REVIEW

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java [hunk 1]
```diff
@@ -124,13 +124,29 @@ protected BaseOperation(OkHttpClient client, Config config, String apiGroup, Str
...
 
+  /**
+   * Separates apiVersion for apiGroup/apiVersion combination.
+   * @param apiVersion  The apiVersion or apiGroup/apiVersion combo.
+   * @return            Just the apiVersion part without the apiGroup.
+     */
+  private static String trimVersion(String apiVersion) {
+    if (apiVersion == null) {
+      return null;
+    } else if (apiVersion.contains("/")) {
```
Comment (jimmidyson) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java]: You can always split regardless of whether it contains a `/`.

**Comment by jimmidyson** [comment]
Why close this?

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java (hunk 1 above)
Comment (iocanel) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java]: If you don't check for contains("/") you'd need to check the length of the array afterwards.
So complexity wise its the same thing. Now performance wise, in the worst case (no / is found) both approach should behave the same. In all other cases its possibly cheaper to check the length of an array compare to check contains. And thank you so much for spotting it :-)

**Comment by iocanel** [comment]
Because, I the same commit is included in an other PR

**Comment by iocanel** [comment]
More specifically: #519

**Pull request closed without merging**